### PR TITLE
Delete non-exec-based agent factories

### PR DESCRIPTION
**Pull request by jglick**
After #18 there is not much reason to continue to support the in-JVM modes, and they are a maintenance burden, so deleting them.

### CODE REVIEW

**Comment by alecharp** [comment]
I believe this is the way to go. I wonder if some kind of migration is required for this. What do you think @jglick ?

**Comment by jglick** [comment]
I am not sure what migration is possible. As before, if you have the `ssh-agent` binary installed on your agent, the plugin will work. If you do not, we cannot do anything other than print an error message, which I think it already does (though as noted in the TODO comment, we could probably tune the logging behavior now).

**Comment by alecharp** [comment]
I was thinking about some other plugin to use, other connection details to prefer.

**Comment by jglick** [comment]
> some other plugin to use

I did note in docs the simpler alternative using `withCredentials` applicable to many use cases.

**Review by alecharp — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -152,13 +132,10 @@
...
     <plugins>
       <plugin>
-        <!-- critical because sshd needs to be loaded first, otherwise will just bail out -->
-        <!-- https://www.jenkins.io/doc/developer/plugin-development/dependencies-and-class-loading/ -->
         <groupId>org.jenkins-ci.tools</groupId>
         <artifactId>maven-hpi-plugin</artifactId>
         <extensions>true</extensions>
         <configuration>
-          <pluginFirstClassLoader>true</pluginFirstClassLoader>
           <compatibleSinceVersion>1.5</compatibleSinceVersion>
```
Comment (daniel-beck) [inline pom.xml:139]: Probably should have had this updated?

Comment (alecharp) [reply pom.xml:139]: we can for the next release. from https://github.com/jenkinsci/ssh-agent-plugin/pull/48#issuecomment-805759155, it shouldn't matter for `1.22` no?

Comment (jglick) [reply pom.xml:139]: IIUC `compatibleSinceVersion` is only relevant for settings changes, and there are none here.

Comment (daniel-beck) [reply pom.xml:139]: Doesn't this break existing configurations that rely on this feautre? It's not _exactly_ configuration compatibility in the strict sense, but I don't think anyone whose jobs we break will care about that distinction.

Comment (jglick) [reply pom.xml:139]: Whatever you think appropriate.

Comment (alecharp) [reply pom.xml:139]: If we have an agreement here, I need to cut a release really soon because I already published `1.22`

**Comment by daniel-beck** [comment]
This change is breaking today's core security release on release.ci 🎉

**Pull request merged**